hazelcast/src/main/java/com/hazelcast/jet/impl/execution/ExecutionContext.java

```diff
@@ -237,33 +268,45 @@ public CompletableFuture<Void> beginExecution(TaskletExecutionService taskletExe
      * Complete local execution. If local execution was started, it should be
      * called after execution has completed.
      */
-    public void completeExecution(Throwable error) {
+    public CompletableFuture<Void> completeExecution(Throwable error) {
         assert executionFuture == null || executionFuture.isDone()
                 : "If execution was begun, then completeExecution() should not be called before execution is done.";
 
         if (!executionCompleted.compareAndSet(false, true)) {
-            return;
+            return completedFuture(null);
         }
         for (Tasklet tasklet : tasklets) {
             try {
                 tasklet.close();
             } catch (Throwable e) {
                 logger.severe(jobNameAndExecutionId()
-                              + " encountered an exception in Processor.close(), ignoring it", e);
+                        + " encountered an exception in Processor.close(), ignoring it", e);
             }
         }
 
         JobClassLoaderService jobClassloaderService = jetServiceBackend.getJobClassLoaderService();
         ClassLoader classLoader = jobClassloaderService.getClassLoader(jobId);
+        List<CompletableFuture<Void>> futures = new ArrayList<>(vertices.size());
+
+        ExecutorService offloadExecutor = nodeEngine.getExecutionService().getExecutor(JOB_OFFLOADABLE_EXECUTOR);
         doWithClassLoader(classLoader, () -> {
             for (VertexDef vertex : vertices) {
-                try {
-                    ClassLoader processorCl = isLightJob ?
-                            null : jobClassloaderService.getProcessorClassLoader(jobId, vertex.name());
-                    doWithClassLoader(processorCl, () -> vertex.processorSupplier().close(error));
-                } catch (Throwable e) {
-                    logger.severe(jobNameAndExecutionId()
-                                  + " encountered an exception in ProcessorSupplier.close(), ignoring it", e);
+                ProcessorSupplier processorSupplier = vertex.processorSupplier();
+                RunnableEx closeAction = () -> {
+                    try {
+                        ClassLoader processorCl = isLightJob ?
+                                null : jobClassloaderService.getProcessorClassLoader(jobId, vertex.name());
+                        doWithClassLoader(processorCl, () ->  processorSupplier.close(error));
+                    } catch (Throwable e) {
+                        logger.severe(jobNameAndExecutionId()
+                                + " encountered an exception in ProcessorSupplier.close(), ignoring it", e);
+                    }
```

Comment:
Here we catch exceptions from `ProcessorSupplier#close` so the asserts in MockPS won't do anything. This is NOT new issue introduced in this PR, but let's fix it (in the test ...).